src/fe/fe_xyz.C

```diff
@@ -700,6 +700,9 @@ void FEXYZ<Dim>::init_shape_functions(const std::vector<Point> & qp,
 
   }
 #endif // ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
+
+  if (this->calculate_dual)
+    this->init_dual_shape_functions(n_approx_shape_functions, n_qp);
```

Comment:
Since you are adding the initializations, do we need to add this to `InfFE<Dim,T_radial,T_map>::init_shape_functions` as well?

Reply:
I don't know much about infinite elements; if dual has meaning there, then I will add it.

Reply:
dual has meaning there, but not the *same* meaning; you can't use the L2 norm anymore without some kind of weighting function.  It should be fine to have an attempt to calculate_dual there end up at libmesh_not_implemented() for now.